### PR TITLE
nfs-ganesha: put flavor after chacra_endpoint

### PR DESCRIPTION
**Pull request by alimaredia**
Signed-off-by: Ali Maredia <amaredia@redhat.com>

### CODE REVIEW

File: nfs-ganesha/build/build_deb [hunk 1]
```diff
@@ -146,7 +146,7 @@ EOF
     $VENV/chacractl repo update ${chacra_endpoint}
```
Comment (andrewschoen) [inline nfs-ganesha/build/build_deb]: You need to add ``/flavors/${FLAVOR}`` here as well.  Actually, I think with debian you can just keep the flavors bit of the url in ``chacra_endpoint``.

File: nfs-ganesha/build/build_rpm [hunk 2]
```diff
@@ -103,7 +103,7 @@ PACKAGE_MANAGER_VERSION="$RPM_VERSION-$RPM_RELEASE"
 if [ "$THROWAWAY" = false ] ; then
     # push binaries to chacra
     find $WORKSPACE/dist/SRPMS | grep rpm | $VENV/chacractl binary ${chacra_flags} create ${chacra_endpoint}/source
```
Comment (andrewschoen) [inline nfs-ganesha/build/build_rpm]: you'll want to add ``/flavors/${FLAVOR}`` here as well

File: nfs-ganesha/build/build_rpm [hunk 3]
```diff
@@ -122,7 +122,7 @@ EOF
     $VENV/chacractl repo update ${chacra_endpoint}
```
Comment (andrewschoen) [inline nfs-ganesha/build/build_rpm]: you'll want to add ``/flavors/${FLAVOR}`` here as well

File: nfs-ganesha/build/build_rpm [hunk 4]
```diff
@@ -117,12 +117,12 @@ if [ "$THROWAWAY" = false ] ; then
 }
 EOF
     # post the json to repo-extra json to chacra
-    curl -X POST -H "Content-Type:application/json" --data "@$WORKSPACE/repo-extra.json" -u $CHACRACTL_USER:$CHACRACTL_KEY ${chacra_url}/repos/${chacra_endpoint}/extra/
+    curl -X POST -H "Content-Type:application/json" --data "@$WORKSPACE/repo-extra.json" -u $CHACRACTL_USER:$CHACRACTL_KEY ${chacra_url}/repos/${chacra_endpoint}/extra/flavors/${FLAVOR}
```
Comment (alimaredia) [inline nfs-ganesha/build/build_rpm]: @andrewschoen is this line, along with the corresponding line in build_deb, supposed to have `flavor/${flavor}` added too?

Comment (andrewschoen) [inline nfs-ganesha/build/build_rpm]: yes, but it needs to come before ``/extra``.   ``ceph-dev-build`` is the best job to look at for examples of chacra integration, see these couple lines here:

https://github.com/ceph/ceph-build/blob/master/ceph-dev-build/build/build_rpm#L174-L176

File: nfs-ganesha/build/build_rpm [hunk 5]
```diff
@@ -117,12 +118,12 @@ if [ "$THROWAWAY" = false ] ; then
 }
 EOF
     # post the json to repo-extra json to chacra
-    curl -X POST -H "Content-Type:application/json" --data "@$WORKSPACE/repo-extra.json" -u $CHACRACTL_USER:$CHACRACTL_KEY ${chacra_url}/repos/${chacra_endpoint}/extra/
+    curl -X POST -H "Content-Type:application/json" --data "@$WORKSPACE/repo-extra.json" -u $CHACRACTL_USER:$CHACRACTL_KEY ${chacra_url}repos/${chacra_repo_endpoint}/extra/
```
Comment (alimaredia) [inline nfs-ganesha/build/build_rpm]: @andrewschoen one thing I noticed was that when the equivalent curl command is run in ceph-dev-build, what I have as `${chacra_url}/repo` is `${chacra_url}repo` (https://github.com/ceph/ceph-build/blob/master/ceph-dev-build/build/build_deb#L107) is this a typo/does it make a difference?

Comment (andrewschoen) [inline nfs-ganesha/build/build_rpm]: It depends on how you've defined ``$chacra_url``, did you include a trailing slash?

Comment (alimaredia) [inline nfs-ganesha/build/build_rpm]: I'm doing the same thing (https://github.com/ceph/ceph-build/blob/master/nfs-ganesha/build/setup#L54) that ceph-dev-build is doing (https://github.com/ceph/ceph-build/blob/master/ceph-dev-build/build/setup_deb#L109), (https://github.com/ceph/ceph-build/blob/master/ceph-dev-build/build/setup_rpm#L117)

**Pull request merged**